pkg/apis/v1alpha1/serviceexport.go

```diff
@@ -88,3 +92,133 @@ type ServiceExportList struct {
 	// +listType=set
 	Items []ServiceExport `json:"items"`
 }
+
+// ServiceExportConditionType is a type of condition associated with ServiceExport.
+// This type should be used with the ServiceExportStatus.Conditions
+// field.
+type ServiceExportConditionType string
+
+// ServiceExportConditionReason defines the set of reasons that explain
+// why a particular conflict condition type has been raised.
+type ServiceExportConditionReason string
+
+const (
+	// ServiceExportConditionValid is true when the Service Export is valid. This does
+	// not indicate whether or not the configuration has been exported to
+	// a control plane / data plane.
+	//
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "Valid"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "NoService"
+	// * "InvalidServiceType"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	ServiceExportConditionValid ServiceExportConditionType = "Valid"
+
+	// ServiceExportReasonValid is used with the "Valid" condition when the
+	// condition is True.
+	ServiceExportReasonValid ServiceExportConditionReason = "Valid"
+
+	// ServiceExportReasonNoService is used with the "Valid" condition when
+	// the associated Service does not exist.
+	ServiceExportReasonNoService ServiceExportConditionReason = "NoService"
+
+	// ServiceExportReasonInvalidServiceType is used with the "Valid"
+	// condition when the associated Service has an invalid type
+	// (per the KEP at least the ExternalName type).
+	ServiceExportReasonInvalidServiceType ServiceExportConditionReason = "InvalidServiceType"
+)
+
+const (
+	// ServiceExportConditionExported is true when the service is exported to some
+	// control plane or data plane.
+	//
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "Exported"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "Invalid"
+	// * "Pending"
+	//
+	// Possible reasons for this condition to be Unknown are:
+	//
+	// * "Pending"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	ServiceExportConditionExported ServiceExportConditionType = "Exported"
+
+	// ServiceExportReasonExported is used with the "Exported" condition
+	// when the condition is True.
+	ServiceExportReasonExported ServiceExportConditionReason = "Exported"
```

Comment:
How about `Success` so we don't duplicate the type name?

Reply:
Duplicating the type name is actually quite typical for the similar status conditions in Gateway API.

Reply:
Perhaps but the way Gateway API does things doesn't necessarily mean it's the best way ... 😄

Reply:
My 2 cents is that our usage of conditions was probably not that great so far (at least for Cilium) and defaulting to GatewayAPI things except if we find good reasons to not do it on a case by case basis. And in that case both options reasonable enough :man_shrugging:.